src/features/Auth/types.ts

```diff
@@ -0,0 +1,29 @@
+export interface IUser {
+    id: number;
+    email: string;
+    password: string;
+}
+
+export interface InitialState {
+    user: IUser;
+    password: string;
+    email: string;
+    isAuth: boolean;
+    isLoading: boolean;
+}
+
+export interface AuthActions  {
+    setAuth: (bool: boolean) => void;
+    setUser: (user: IUser) => void;
+    setPassword: (password: string) => void,
+    setEmail: (password: string) => void,
+    setIsLoading: (bool: boolean) => void;
+    login: (email: string, password: string) => Promise<undefined | boolean>;
+    logout: (token: string) => Promise<void>;
+    registration :(email: string, password: string, secretKey:string) => Promise<undefined | boolean>;
+}
+
+export interface authData {
```

Comment:
Интерфейсы принято называт с большой буквы 
```suggestion
export interface AuthData {
```